File: esp32_arm/src/MyApps/gobot_house/HouseMap.cpp
# include "HouseMap.h"
#include <math.h>

/*

	House                Y+
						 ^
			 r0          |                                                                     \ beta  
		  r1     d0      |                                                                    \  /
		r2      d1       |                                                                     \/      
	  r3       d3  d2    |                                                                   /  alpha
	  -----------------(0,0)------Neck----------Head    --> X+                               --------------------------------->  Alpha,Beta (0,0)
	  r4      d4  d5     |
	   r5       d6       |
		 r6     d7       |
		  r7             |

*/

void HouseMap::Init(){
    this->neck.Y = 0;
    // this->neck.X = 55.0;
    this->neck.X = 65.0;
    this->head.Y = 0;
    this->head.X = 148.93;

    this->rooms[0].X = -78.79;
    this->rooms[0].Y = 126.38;
    this->rooms[1].X = -98.18;
    this->rooms[1].Y = 95.34;
    this->rooms[2].X = -111.9;
    this->rooms[2].Y = 61.4;
    this->rooms[3].X = -119.51;
    this->rooms[3].Y = 25.59;

    this->rooms[4].X = this->rooms[3].X;
    this->rooms[4].Y = - this->rooms[3].Y;
    this->rooms[5].X = this->rooms[2].X;
    this->rooms[5].Y = - this->rooms[2].Y;
    this->rooms[6].X = this->rooms[1].X;
    this->rooms[6].Y = - this->rooms[1].Y;
    this->rooms[7].X = this->rooms[0].X;
    this->rooms[7].Y = - this->rooms[0].Y;

    // this->gates[0].X = -10.37;
    // this->gates[0].Y = 59.07;
    // this->gates[1].X = -21.39;
    // this->gates[1].Y = 56.12;
    // this->gates[2].X = -44.28;
    // this->gates[2].Y = 40.67;
    // this->gates[3].X = -57.42;
    // this->gates[3].Y = 17.95;


    this->gates[4].X = -3;
    this->gates[4].Y = -65;
    this->gates[5].X = -27;
    this->gates[5].Y = -59;
    this->gates[6].X = -49;
    this->gates[6].Y = -42;
    this->gates[7].X = -62;
    this->gates[7].Y = -19;

    this->gates[3].X = -62;
    this->gates[3].Y = 19;
    this->gates[0].X = this->gates[3].X;
    this->gates[0].Y = this->gates[3].Y;
    this->gates[1].X = this->gates[3].X;
    this->gates[1].Y = this->gates[3].Y;
    this->gates[2].X = this->gates[3].X;
    this->gates[2].Y = this->gates[3].Y;



    this->doors[4].X = -51;
    this->doors[4].Y = -105;
    this->doors[5].X = -67;
    this->doors[5].Y = -89;
    this->doors[6].X = -78;
    this->doors[6].Y = -51;
    this->doors[7].X = -85;
    this->doors[7].Y = -21;
    
    this->doors[0].X = this->doors[4].X;
    this->doors[0].Y = - this->doors[4].Y;
    this->doors[1].X = this->doors[5].X;
    this->doors[1].Y = - this->doors[5].Y;
    this->doors[2].X = this->doors[6].X;
    this->doors[2].Y = - this->doors[6].Y;
    this->doors[3].X = this->doors[7].X;
    this->doors[3].Y = - this->doors[7].Y;
}

File: gogame/house_map.py
import enum

from Pylib.robot_map import MapSite

class HouseMapSite_Catalog(enum.Enum):
    ROOM = 1,
    DOOR = 2,
    HEAD = 3,
    NECK = 4,
    PARKING = 5, 

class HouseMapSites():
    def __init__(self) -> None:
        self.head = MapSite('Head', 148.93, 0)
        self.neck = MapSite('Neck', 55.0, 0)
        self.parking = MapSite('Parking', 55.0, -2)

        self.doors = [MapSite]
        self.doors.append (MapSite('Door_0',-10.37,  59.07))
        self.doors.append (MapSite('Door_1',-21.39, 56.12))
        self.doors.append (MapSite('Door_2',-44.28, 40.67))
        self.doors.append (MapSite('Door_3',-57.42, 17.95))
        self.doors.append (MapSite('Door_4',-57.42, -17.95))
        self.doors.append (MapSite('Door_5',-44.28, -40.67))
        self.doors.append (MapSite('Door_6',-21.39, -56.12))
        self.doors.append (MapSite('Door_7',-10.37, -59.07))
        
        self.rooms = [MapSite]
        self.rooms.append(MapSite('Room_0', -78.79, 126.38))
        self.rooms.append(MapSite('Room_1', -98.18, 95.3))
        self.rooms.append(MapSite('Room_2', -111.9, 61.4))
        self.rooms.append(MapSite('Room_3', -119.51, 25.59))
        self.rooms.append(MapSite('Room_4', -119.51, -25.59))
        self.rooms.append(MapSite('Room_5', -111.9, -61.4))
        self.rooms.append(MapSite('Room_6', -98.18, -95.2))
        self.rooms.append(MapSite('Room_7', -78.79, -126.38))

    def GetSingleSite(self, cat:HouseMapSite_Catalog, index:int) -> MapSite:
        if cat == HouseMapSite_Catalog.HEAD:
            return self.head
        elif cat == HouseMapSite_Catalog.NECK:
            return self.neck
        elif cat == HouseMapSite_Catalog.DOOR:
            return self.doors[index]
        elif cat == HouseMapSite_Catalog.ROOM:
            return self.rooms[index]
        elif cat == HouseMapSite_Catalog.PARKING:
            return self.parking
        else:
            print("[Error] HouseMapSites.GetSingleSite()  cat=, index= ", cat, index)


